meteor/client/ui/SegmentTimeline/SegmentTimelinePart.tsx

```diff
@@ -428,533 +454,587 @@ const CARRIAGE_RETURN_ICON = (
-										liveLineHistorySize={this.props.liveLineHistorySize}
-										livePosition={this.props.livePosition}
-										onContextMenu={this.props.onContextMenu}
-										onFollowLiveLine={this.props.onFollowLiveLine}
-										onPieceClick={this.props.onPieceClick}
-										onPieceDoubleClick={this.props.onPieceDoubleClick}
-										scrollLeft={this.props.scrollLeft}
-										scrollWidth={this.props.scrollWidth}
-										relative={this.props.relative}
-										mediaPreviewUrl={
-											ensureHasTrailingSlash(this.props.studio.settings.mediaPreviewsUrl + '' || '') || ''
-										}
-										layer={layer}
-										sourceLayers={sourceLayers}
-										segment={this.props.segment}
-										part={part}
-										playlist={this.props.playlist}
-										studio={this.props.studio}
-										startsAt={SegmentTimelinePart0.getPartStartsAt(this.props) || this.props.part.startsAt || 0}
-										duration={SegmentTimelinePart0.getPartDuration(this.props, this.state.liveDuration)}
-										expectedDuration={SegmentTimelinePart0.getPartExpectedDuration(this.props)}
-										isLiveLine={this.props.playlist.currentPartInstanceId === part.instance._id}
-										isNextLine={this.props.playlist.nextPartInstanceId === part.instance._id}
-										timeScale={this.props.timeScale}
-										autoNextPart={this.props.autoNextPart}
-										liveLinePadding={SegmentTimelinePart0.getLiveLineTimePadding(this.props.timeScale)}
-										indexOffset={currentIndex}
-									/>
-								)
-							}
-						})
-				}
+	getLayerStyle() {
+		// this.props.part.expectedDuration ||
+		const actualPartDuration = SegmentTimelinePartClass.getPartDuration(this.props, this.state.liveDuration)
+		const partDuration = this.props.cropDuration
+			? Math.min(this.props.cropDuration, actualPartDuration)
+			: actualPartDuration
+		if (this.props.relative) {
+			return {
+				width: ((partDuration / (this.props.totalSegmentDuration || 1)) * 100).toString() + '%',
+				// width: (Math.max(this.state.liveDuration, this.props.part.duration || this.props.part.expectedDuration || 3000) / (this.props.totalSegmentDuration || 1) * 100).toString() + '%',
+				willChange: this.state.isLive ? 'width' : undefined,
 			}
-
-			getFutureShadeStyle = () => {
-				return {
-					width:
-						Math.min(
-							Math.max(
-								0,
-								(this.props.livePosition || 0) +
-									SegmentTimelinePart0.getLiveLineTimePadding(this.props.timeScale) -
-									(this.props.part.instance.part.expectedDuration || this.props.part.renderedDuration || 0)
-							),
-							SegmentTimelinePart0.getLiveLineTimePadding(this.props.timeScale)
-						) *
-							this.props.timeScale +
-						'px',
-				}
+		} else {
+			return {
+				minWidth: Math.floor(partDuration * this.props.timeScale).toString() + 'px',
+				// minWidth: (Math.max(this.state.liveDuration, this.props.part.duration || this.props.part.expectedDuration || 3000) * this.props.timeScale).toString() + 'px',
```

Comment:
Remove dead code or add a comment saying why it should be preserved

Reply:
Agreed. This was a leftover from to make sure the new implementation covered all bases.